osquery/tables/yara/yara_utils.cpp

```diff
@@ -111,6 +132,50 @@ Status compileSingleFile(const std::string& file, YR_RULES** rules) {
   return Status::success();
 }
 
+/**
+ * Compile yara rules from string and load it into rule pointer.
+ */
+Status compileFromString(const std::string& rule_defs, YR_RULES** rules) {
+  YR_COMPILER* compiler = nullptr;
+
+  auto result = yr_compiler_create(&compiler);
+  if (result != ERROR_SUCCESS) {
+    return Status(1, " - Could not create compiler: " + std::to_string(result));
```

Comment:
Prefer the new `Status::failure` API, and please do not include the ` - ` prefix in the message. If you want this "pretty printed" somewhere, do that formatting in the status log line.